Extension/src/LanguageServer/configurations.ts

```diff
@@ -737,31 +738,51 @@ export class CppProperties {
 
     private resolveAndSplit(paths: string[] | undefined, defaultValue: string[] | undefined, env: Environment, glob = false): string[] {
         const result: string[] = [];
-        if (paths) {
-            paths = this.resolveDefaults(paths, defaultValue);
+        if (paths === undefined) {
+            return result;
+        }
+        paths = this.resolveDefaults(paths, defaultValue);
+        if (!glob) {
             paths.forEach(entry => {
                 const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
                 result.push(...entries);
             });
-        }
-        if (!glob) {
             return result;
         }
 
         const globResult: string[] = [];
```

Comment:
`const result: string[] = [];` can be used for both cases and `const globResult: string[] = [];` can be removed. Since if `glob` is true, then `result` won't be used for the first case.